staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -93,10 +93,10 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 		ae := request.AuditEventFrom(ctx)
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
-		if admit != nil && admit.Handles(admission.Create) {
-			userInfo, _ := request.UserFrom(ctx)
-
-			err = admit.Admit(admission.NewAttributesRecord(obj, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Create, userInfo))
+		userInfo, _ := request.UserFrom(ctx)
```

Comment:
I'm considering if we should move the mutation after BeforeCreate(), which changes the metadata field and clears the status.

Similarly if we should move the mutation for update to after BeforeUpdate().

Reply:
> I'm considering if we should move the mutation after BeforeCreate(), which changes the metadata field and clears the status.
> 
> Similarly if we should move the mutation for update to after BeforeUpdate().

I think that's a more significant change.  Particular since `BeforeCreate` and `BeforeUpdate` do `Validate`.  Perhaps as a followup, but not as a first step until we've considered repercussions like the validation one?

Reply:
Sorry that was typed in a hurry. I meant to say it's worth considering plumbing the admission mutation **into** `BeforeCreate`/`BeforeUpdate`, after the metadata changes and `PrepareForCreate`/`PrepareForUpdate`, but **before** the validation.
I quickly scanned through our existing mutation plugins, they don't care about the fields changed in `BeforeCreate` or `BeforeUpdate`, so the order we have in this PR is ok.